09-graphs-p2/01-graphs-review.md

```diff
@@ -0,0 +1,325 @@
+
+The main reason the DFS algorithm is not typically used to find the shortest path is because DFS is not a "greedy" algorithm. The algorithm is not designed to account for changes in logic based on the data it encounters, which is what is needed for finding the shortest path in a graph.
+
+## !callout
+A greedy algorithm is an approach for solving a problem by selecting the best option available at the moment. A greedy algorithm never reverses the earlier decision even if the choise is wrong. An algorithm which is considered greedy may not produce the best result for all problems.
+## !end-callout
+
+### Why BFS (kinda) allows for shortest path of an unweighted graph
+
+By itself, BFS does *not* allow for finding the shortest path of an unweighted graph. In order to retrieve the shortest path using the BFS algorithm, one would need to modify the algorithm to store the current shortest distance to the target node as well as the preceding node in the shortest path. This is essentially what Djikstra's algorithm does, which we will learn more about in the next section.
+
+## BFS, DFS, or Either
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: multiple-choice
+* id: e012797e-0060-4484-b3b4-245fa4152228
+* title: BFS, DFS, Either, or Neither?
+* points: 0
+
+##### !question
+
+If we are looking for a node in a graph that is many levels deeper than the starting node, is BFS, DFS, Either, or Neither the best option?
+
+##### !end-question
+
+##### !options
+
+a| Breadth First Search
+b| Depth First Search
+c| Either
+d| Neither
+
+##### !end-options
+
+##### !answer
+
+b|
+
+##### !end-answer
+
+##### !explanation
+
+The correct answer is B because it is faster and more efficent to reach nodes deeper in the graph with Depth First Search.
+
+##### !end-explanation 
+
+### !end-challenge
+
+### !challenge
+
+* type: multiple-choice
+* id: b8569433-61d6-48e0-afa2-5b28e3e9facf
+* title: BFS, DFS, Either, or Neither?
+* points: 0
+
+##### !question
+
+If we are looking to find all the connected nodes in a graph, is BFS, DFS, Either, or Neither the best option?
+
+##### !end-question
+
+##### !options
+
+a| Breadth First Search
+b| Depth First Search
+c| Either
+d| Neither
+
+##### !end-options
+
+##### !answer
+
+a|
+
+##### !end-answer
+
+##### !explanation
+
+The correct answer is A because BFS allows us to find which nodes are connected to one another in an efficient manner. We would not necessarily want to use DFS to find all connected nodes because the algorithm does not lend itself to finding connected nodes efficiently. 
+
+For instance, if we are looking to see if a node 2 levels away from the starting node is connected, the DFS algorithm may need to run for a much longer time to find that connected node that is placed nearby in the graph in comparison to the BFS algorithm which will find the connected node much sooner.
```

Comment:
The question and the explanation don't totally align for me.

An overall traversal of the graph from a start node will return the list of connected nodes for both BFS and DFS and I believe both have the same time complexity with DFS being on average slightly faster.

Your explanation does make sense with your example of looking for a node 2 levels away, but with the way the question is phrased I was thinking more about which option would allow me to find the set of all connected nodes.

Maybe rephrasing the question to check if two nodes are connected?